test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/LocalClusterSpec.java

```diff
@@ -212,7 +212,12 @@ public Path getConfigDir() {
         }
 
         public boolean isSecurityEnabled() {
-            return Boolean.parseBoolean(getSetting("xpack.security.enabled", getVersion().onOrAfter("8.0.0") ? "true" : "false"));
+            return Boolean.parseBoolean(
+                getSetting(
+                    "xpack.security.enabled",
+                    getVersion().equals(Version.fromString("0.0.0")) || getVersion().onOrAfter("8.0.0") ? "true" : "false"
```

Comment:
This fix makes sure to enable security if using the magic version number `0.0.0` (we're always > 8.0.0 in this case as we won't be backporting this as far)

Reply:
This probably affects our serverless tests too. Have we verified this change works well with serverless?